### PR TITLE
Defrag etcd data

### PR DESCRIPTION
**Pull request by xrstf**
**What this PR does / why we need it**:
We found that defragmenting the etcd data directories makes them use much less RAM, CPU and improve backup performance considerably. For some clusters with 500MB data directories per node, the data shrunk down to less than 10MB. In general, the `defrag` operations takes less than 5 seconds, usually less than 1 second.

This PR adds a new CronJob to a cluster's namespace which will defrag etcd once every 3 hours. While defragmenting, etcd cannot serve any requests. This is why the script in this PR's cronjob waits a couple of seconds between nodes.

The PR seems rather large because it not only defines a new CronJob, but also teaches the cluster controller how to handle CronJobs in the first place. Most of this is simply copied from the existing resource logic.

**Special notes for your reviewer**:
In the cluster_controller.go, line 902, I have no idea if this is really meaningful here and makes sense. I copied it from the code handling StatefulSets and it works, but you might want to take a closer look.

**Release note**:
```release-note
etcd in user cluser namespaces is defragmented every 3 hours
```

### CODE REVIEW

File: api/pkg/controller/cluster/resources.go [hunk 1]
```diff
@@ -82,22 +83,27 @@ func (cc *Controller) ensureResourcesAreDeployed(cluster *kubermaticv1.Cluster)
 		return err
 	}
 
-	// check that all ServerClientConfigsConfigMap's are available
+	// check that all ServerClientConfigsConfigMaps are available
```
Comment (mrIncompetent) [inline api/pkg/controller/cluster/resources.go]: ServerClientConfigsConfigMaps -> ConfigMaps

**Comment by mrIncompetent** [comment]
Please add handling for the new creator in the image-loader cmd:
https://github.com/kubermatic/kubermatic/blob/master/api/cmd/image-loader/main.go#L158

**Comment by xrstf** [comment]
image-loader is updated.

**Comment by mrIncompetent** [comment]
load_files tests are missing, apart for that lgtm

**Review by mrIncompetent — DISMISSED** [review]

**Review by mrIncompetent — APPROVED** [review]

**Pull request merged**